lib/src/Weather/WeatherReport.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright(c), Philips Medical Systems Nederland B.V., IGT-Devices
+
+#include <Weather/IWeatherSensor.h>
+#include <Weather/WeatherReport.h>
+
+#include <string>
+
+namespace Weather {
+
+namespace {
+constexpr auto kSunnyDay  = "Sunny Day";
+constexpr auto kRainyDay  = "Rainy Day";
+constexpr auto kStormyDay = "Alert, Stormy with heavy rain";
+} // namespace
+
+WeatherReport report(const IWeatherSensor& sensor)
+{
+    const auto precipitation = sensor.precipitation();
+
+    if (precipitation < 20) return WeatherReport::Sunny;
```

Comment:
The "property of the weather" is nicely visible here in 3 lines of code: _low precipitation would mean sun, high would mean rain or storm, depending on the wind_
But this is not so visible in the test. Pls see comment on the test too.
Why should the "property" be visible in the test? Consider this example: Let's say an AI algorithm may do a better job, maybe reporting in natural language. Still, the properties of the weather are the same. The forecast needs to have one of the words `sun`, `rain` or `storm`. So even when the implementation changes, this kind of test would not have to change.